src/Transition.ts

```diff
@@ -36,24 +35,34 @@ export const Transition: Component<TransitionProps> = props => {
   const [s2, set2] = createSignal<Element | undefined>();
   const resolved = children(() => props.children);
   const name = props.name || "s";
-  props = mergeProps(
-    {
-      name,
-      enterActiveClass: name + "-enter-active",
-      enterClass: name + "-enter",
-      enterToClass: name + "-enter-to",
-      exitActiveClass: name + "-exit-active",
-      exitClass: name + "-exit",
-      exitToClass: name + "-exit-to"
-    },
-    props
-  );
+
+  let enterActiveClass = name + "-enter-active"
+  let enterClass = name + "-enter"
+  let enterToClass = name + "-enter-to"
+  let exitActiveClass = name + "-exit-active"
+  let exitClass = name + "-exit"
+  let exitToClass = name + "-exit-to"
+
   const { onBeforeEnter, onEnter, onAfterEnter, onBeforeExit, onExit, onAfterExit } = props;
+
+  createEffect(() => {
+    // effect if the name of animation changes
+    const name = props.name || "s";
+
+    // declare classes with new name property
+    enterActiveClass = name + "-enter-active"
+    enterClass = name + "-enter"
+    enterToClass = name + "-enter-to"
+    exitActiveClass = name + "-exit-active"
+    exitClass = name + "-exit"
+    exitToClass = name + "-exit-to"
+  })
```

Comment:
I think that removing the `mergeProps` is a good call—it's not necessary for anything.
Deriving the class names via effect will work for most cases, but shouldn't it be derived with `createComputed`, or better `createMemo` instead?
The animation is being handled by `createComputed` on line 128 which will run before this `createEffect` if both were to be triggered at the same time.
```suggestion
const classnames = createMemo(() => {
  const name = props.name || "s";

  return {
    enterActiveClass: name + "-enter-active",
    enterClass: name + "-enter",
    enterToClass: name + "-enter-to",
    exitActiveClass: name + "-exit-active",
    exitClass: name + "-exit",
    exitToClass: name + "-exit-to",
  };
});
```

Reply:
Yes, this is a better idea.
I checked it and it worked.
I close this PR and open another adding your suggestion and some other changes.